enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,359 @@
+  I can configure appropriate firewall rules.
+  I want the default AWS Load Balancer that they use (NLB) for their router to use these EIPs.
+
+### Goals
+- Users are able to use EIP for a NLB Ingress Controller.
+- Users are able to create a NLB default Ingress Controller during install time with the EIPs specified during install.
+- Any existing ingress controller should be able to be reconfigured to use EIPs. 
+
+### Non-Goals
+
+- Creation of EIPs in AWS.
+- Keep a track of changes in the EIP in the AWS environment.
+- Static IP usage with NLBs for API server.
+
+## Proposal
+
+In this enhancement we would be adding API fields in the installer and the ingress controller CRD
+to set the AWS EIP for the NLB Ingress Controller. The cluster ingress operator will then create 
+the ingress controller and get the EIP assigned to the service LoadBalancer with the annotation 
+`service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible for deploying a
+cluster. A cluster administrator also has the rights to modify the cluster level components.
+
+#### Set EIP using an installer:
+1. Cluster administrator creates adds the following in the install-config.yaml
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-west-2
+    lbType: NLB
+      networkLoadBalancer:
+        eip-allocations: eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted> 
+    subnets:
+    - subnet-1
+    - subnet-2
+    - subnet-3
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```  
+2. The installer will create the cluster ingress config object as follows:
+```yaml
+ oc edit ingress.config/cluster -o yaml
+ # Please edit the object below. Lines beginning with a '#' will be ignored,
+ # and an empty file will abort the edit. If an error occurs while saving this file will be
+ # reopened with the relevant failures.
+ #
+ apiVersion: config.openshift.io/v1
+ kind: Ingress
+ metadata:
+   creationTimestamp: "2022-08-30T01:03:17Z"
+   generation: 11
+   name: cluster
+   resourceVersion: "60888"
+   uid: de89cc95-449e-4ae5-a837-f4171fe61e36
+ spec:
+   domain: apps.testnlbapi.devcluster.openshift.com
+   loadBalancer:
+     platform:
+       aws:
+         type: NLB
+           networkLoadBalancer:
+             eip-allocations: eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>,eipalloc-<redacted>
+       type: AWS
+ status:
+   componentRoutes:
+```
+3. Ingress Operator will create the default ingress controller with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`.
+
+#### Create an Ingress Controller using Ingress Controller CR and assign the EIP allocation mentioned in the Ingress Controller CR.
```

Comment:
Is this a part of the previous workflow?  It would be better to group them together if so.

Reply:
This is a different workflow. Day Two we can say i.e after installation.